playbooks/adhoc/docker_storage_to_overlay2/docker_storage_to_overlay2.yml

```diff
@@ -0,0 +1,197 @@
+        reason: "a 'storage modify' bugfix"
+
+      # Ensure XFS volumes are mounted with the pquota option.
+      container-storage-setup:
+        minimum: 0.7.0-1
+        reason: "pquota mount option"
+
+      # Ensure we have the backported overlay2.size storage option.
+      docker:
+        minimum: 1.12.6-58
+        reason: "overlay2.size storage option"
+
+  tasks:
+
+  - name: Install or update required packages
+    package:
+      name: "{{ item.key }}"
+      state: latest
+    with_dict: "{{ package_requirements }}"
+
+  - name: Start docker.service
+    service:
+      name: docker
+      state: started
+
+  - name: Capture output of "docker info"
+    command: docker info
+    register: docker_info_output
+    changed_when: False
+    ignore_errors: yes
+
+  - name: Isolate storage driver
+    set_fact:
+      old_storage_driver: "{{ docker_info_output.stdout | regex_search('(?<=Storage Driver:)\\s*(\\w+)', '\\1', ignorecase=True, multiline=True) }}"
+
+  - name: Transition storage driver to overlay2
+    when: old_storage_driver[0] != "overlay2"
+    block:
+
+    # As of 7.4, SELinux security labels are now supported
+    # on the OverlayFS file system (see RHEL release notes).
+    - name: Check if distribution supports overlayfs (RHEL/CentOS)
+      when: ansible_distribution in ["RedHat", "CentOS"]
+      assert:
+        that: ansible_distribution_version | version_compare("7.4", "ge")
+        msg: "The OverlayFS file system only works with SELinux in {{ ansible_distribution }} 7.4 or later."
+
+    # XXX Is there no easier way to do this?
+    - name: Get {{ item.key }} version
+      command: rpm --query --queryformat="%{version}-%{release}" "{{ item.key }}"
+      args:
+        warn: no
+      register: package_version
+      changed_when: False
+      with_dict: "{{ package_requirements }}"
+
+    - name: Ensure "{{ item.item.key }}" >= "{{ item.item.value.minimum }}"
+      assert:
+        that: item.stdout | version_compare(item.item.value.minimum, "ge")
+        msg: "{{ item.item.key }} >= {{ item.item.value.minimum }} required for {{ item.item.value.reason }}"
+      with_items: "{{ package_version.results }}"
+
+    - name: Check which optional services are available
+      command: systemctl status {{ item }}
+      register: service_available
+      ignore_errors: yes
+      changed_when: False
+      with_items: "{{ optional_service_names }}"
+
+    - name: Add available optional services to service list
+      when: item.rc == 0
+      set_fact:
+        service_names: "{{ service_names + [item.item] }}"
+      with_items: "{{ service_available.results }}"
+
+    - name: Stop services
+      service:
+        name: "{{ item }}"
+        state: stopped
+      with_items: "{{ service_names }}"
+
+    # This step is highly destructive!
+    - name: Remove Docker storage
+      command: atomic storage reset
+
+    - name: Capture storage setup configuration
+      command: cat "{{ storage_setup_path }}"
```

Comment:
Another "nice to have"... Ops likes everything to be in module form if possible, rather than commands. The `slurp` module will do what you need for this task. https://docs.ansible.com/ansible/latest/slurp_module.html